### DISCUSSION TITLE
Copilot Content Exclusion Ruleset Support

### DISCUSSION THREAD

**Discussion by rjwink** [community/community#123790] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

It may be more effective to establish content exclusion via repository custom properties and rulesets.

Example: custom property categorizes a set of repositories as top secret, content exclusion for copilot would be to exclude any repository with the custom property "top secret"

**Comment by nikhilsbuilds** [comment]
You're looking for a more targeted approach to exclude specific content from Copilot's suggestions. By leveraging custom properties and rulesets, you can create a more fine-grained control over what content is excluded.

As you mentioned, creating a custom property like "top secret" and applying it to the relevant repositories can help you exclude specific content from Copilot's suggestions. This approach allows you to categorize your repositories based on their sensitivity level and exclude content accordingly.

Here are some potential benefits of using custom properties and rulesets for content exclusion:

1. **Precision**: By using custom properties, you can create a more precise exclusion criteria, ensuring that only the specific content you want to exclude is removed from Copilot's suggestions.
2. **Flexibility**: Custom properties and rulesets provide a flexible framework for managing content exclusion. You can create multiple custom properties and rulesets to accommodate different scenarios and requirements.
3. **Scalability**: As your repository grows, custom properties and rulesets can help you scale your content exclusion strategy to accommodate new repositories and content types.

To implement this approach, you can follow these general steps:

1. Create a custom property (e.g., "top secret") and apply it to the relevant repositories.
2. Configure a ruleset to exclude content from Copilot's suggestions based on the custom property.
3. Test and refine the ruleset to ensure it accurately excludes the desired content.

By using custom properties and rulesets, you can create a more targeted and effective content exclusion strategy for Copilot.
